pkg/controller/deployment/deployment_controller.go

```diff
@@ -566,152 +662,29 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	// See https://github.com/kubernetes/kubernetes/issues/23211.
 
 	if d.Spec.Paused {
-		return dc.sync(d)
+		return dc.sync(d, rsList, podMap)
 	}
 
 	if d.Spec.RollbackTo != nil {
 		revision := d.Spec.RollbackTo.Revision
-		if d, err = dc.rollback(d, &revision); err != nil {
+		if d, err = dc.rollback(d, rsList, podMap, &revision); err != nil {
 			return err
 		}
 	}
 
-	scalingEvent, err := dc.isScalingEvent(d)
+	scalingEvent, err := dc.isScalingEvent(d, rsList, podMap)
 	if err != nil {
 		return err
 	}
 	if scalingEvent {
-		return dc.sync(d)
+		return dc.sync(d, rsList, podMap)
 	}
 
 	switch d.Spec.Strategy.Type {
 	case extensions.RecreateDeploymentStrategyType:
-		return dc.rolloutRecreate(d)
+		return dc.rolloutRecreate(d, rsList, podMap)
 	case extensions.RollingUpdateDeploymentStrategyType:
-		return dc.rolloutRolling(d)
+		return dc.rolloutRolling(d, rsList, podMap)
 	}
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
-
-// handleOverlap will avoid syncing the newer overlapping ones (only sync the oldest one). New/old is
```

Comment:
I'm all for removing this but technically since we've started handling overlapping replica sets, it's a disruptive change. I was planning on removing this code in 1.8 when we will stop supporting upgrades from 1.5 - the last release w/o owner references for Deployments.